src/Compilers/CSharp/Test/Symbol/Symbols/ExtendedPartialMethodsTests.cs

```diff
@@ -0,0 +1,1430 @@
+
+            var method = (MethodSymbol)comp.GetMembers("C.M1")[0];
+            Assert.True(method.IsPartialDefinition());
+            // PROTOTYPE: it feels like both IsAsync/IsExtern should be shared between
+            // definition+implementation, or neither should.
+            // Currently IsExtern is shared but IsAsync is not.
+            Assert.False(method.IsAsync);
+            Assert.True(method.PartialImplementationPart.IsAsync);
+        }
+
+        [Fact]
+        public void Async_02()
+        {
+            const string text = @"
+using System.Threading.Tasks;
+
+partial class C
+{
+    private partial Task M1();
+    private async partial Task M1() { }
+}
+";
+            var comp = CreateCompilation(text, parseOptions: TestOptions.RegularWithExtendedPartialMethods);
+            comp.VerifyDiagnostics(
+                // (7,32): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread.
+                //     private async partial Task M1() { }
+                Diagnostic(ErrorCode.WRN_AsyncLacksAwaits, "M1").WithLocation(7, 32));
+
+            var method = (MethodSymbol)comp.GetMembers("C.M1")[0];
+            Assert.True(method.IsPartialDefinition());
+            // PROTOTYPE: it feels like both IsAsync/IsExtern should be shared between
+            // definition+implementation, or neither should.
+            // Currently IsExtern is shared but IsAsync is not.
+            Assert.False(method.IsAsync);
+            Assert.True(method.PartialImplementationPart.IsAsync);
+        }
+
+        [Fact]
+        public void Async_03()
+        {
+            const string text = @"
+using System.Threading.Tasks;
+using System;
+
+partial class C
+{
+    private static async Task CompletedTask() { }
+
+    private static partial Task<int> M1();
+    private static async partial Task<int> M1() { await CompletedTask(); return 1; }
+
+    public static async Task Main()
+    {
+        Console.Write(await M1());
+    }
+}
+";
+            var verifier = CompileAndVerify(text, parseOptions: TestOptions.RegularWithExtendedPartialMethods, expectedOutput: "1");
+            verifier.VerifyDiagnostics(
+                // (7,31): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread.
+                //     private static async Task CompletedTask() { }
+                Diagnostic(ErrorCode.WRN_AsyncLacksAwaits, "CompletedTask").WithLocation(7, 31));
+
+            var method = (MethodSymbol)verifier.Compilation.GetMembers("C.M1")[0];
+            Assert.True(method.IsPartialDefinition());
+            // PROTOTYPE: it feels like both IsAsync/IsExtern should be shared between
```

Comment:
>IsAsync/IsExtern should be shared between [](start = 45, length = 41)

Do we have a test with `async` on both parts or `async` on the declaration only?
```C#
private static partial async Task F();
private static partial Task F()
{
    await Task.Yield(); // error on await?
}
```

Reply:
Added the tests.